controller/api/destination/endpoint_translator.go

```diff
@@ -213,14 +263,26 @@ func (et *endpointTranslator) sendClientAdd(set watcher.AddressSet) {
 			err error
 		)
 		if address.Pod != nil {
-			opaquePorts, ok, getErr := getPodOpaquePortsAnnotations(address.Pod)
-			if getErr != nil {
-				et.log.Errorf("failed getting opaque ports annotation for pod: %s", getErr)
+			opaquePorts := make(map[uint32]struct{})
+			id := watcher.PodID{
+				Name:      address.Pod.Name,
+				Namespace: address.Pod.Namespace,
 			}
-			// If the opaque ports annotation was not set, then set the
-			// endpoint's opaque ports to the default value.
-			if !ok {
-				opaquePorts = et.defaultOpaquePorts
+			if ports, ok := set.OpaquePodPorts[id]; ok {
```

Comment:
I think this also gets simpler if we put an opaqueProtocol field on Address.  We'd be able to get rid of all this and simply update toWeightedAddr to take the opaqueProtocol field into account.